### PR TITLE
gymnasium api

### PR DESCRIPTION
**Pull request by luigiberducci**
Updating env api to new gymnasium interface.

Changes:
- f110-env: definition of spaces and their bounds, updating step/reset returns, add slow "rgb_array" render mode.
- unit-test: use the check-env utility from gymnasium
- example: update waypoint-follower to new API

### CODE REVIEW

**Review by hzheng40 — CHANGES_REQUESTED** [review]
LGTM, @nandantumu ?

File: gym/f110_gym/envs/f110_env.py [hunk 1]
```diff
@@ -166,26 +172,56 @@ def __init__(self, **kwargs):
...
         # initiate stuff
-        self.sim = Simulator(self.params, self.num_agents, self.seed, time_step=self.timestep, integrator=self.integrator)
+        self.sim = Simulator(self.params, self.num_agents, self.seed, time_step=self.timestep,
+                             integrator=self.integrator)
         self.sim.set_map(self.map_path, self.map_ext)
 
+        # observation space
+        # NOTE: keep original structure of observation space (dict). just define it as a dict space and define bounds
+        scan_size, scan_range = self.sim.agents[0].scan_simulator.num_beams, self.sim.agents[0].scan_simulator.max_range
+        large_num = 1e5  # large number to avoid unbounded obs space (ie., low=-inf or high=inf)
```
[the remark this answers is not in the document]
Comment (luigiberducci) [reply gym/f110_gym/envs/f110_env.py:197]: Is there any map which requires poses to be beyond +-1e5 meters?

What about the bound of the velocities? Shall we match the simulator limits or keep them like now?

[the remark this answers is not in the document]
Comment (nandantumu) [reply gym/f110_gym/envs/f110_env.py:197]: Let's do 1e30 ? The range of a float is 3.XX x 10^38

**Review by nandantumu — APPROVED** [review]
I made some updates to change the filename of the test, which is great (all PRs should include tests, imo), and updated the large num as per our conversation.

**Pull request merged**